prov/sockets/include/sock.h

```diff
@@ -1211,6 +1218,14 @@ int sock_epoll_wait(struct sock_epoll_set *set, int timeout);
 int sock_epoll_get_fd_at_index(struct sock_epoll_set *set, int index);
 void sock_epoll_close(struct sock_epoll_set *set);
 
+int sock_epoll_map_init(struct sock_epoll_map *emap);
+int sock_epoll_map_destroy(struct sock_epoll_map *emap);
+int sock_epoll_map_add(struct sock_epoll_map *emap, int fd, void *ptr);
+int sock_epoll_map_del(struct sock_epoll_map *emap, int fd);
+void *sock_epoll_map_lookup(struct sock_epoll_map *emap, int fd);
+int sock_epoll_map_signal(struct sock_epoll_map *emap);
+int sock_epoll_map_wait(struct sock_epoll_map *emap);
```

Comment:
This appears generic and not tied specifically to the socket provider.  It would be better moved into general utility code, with a clear definition as to how it can be used.  The use case isn't clear from this patch.  Please see fi_util.h util_map abstraction to see if that provides functionality similar to what you're wanting, but at a higher level.

If you're just wanting to improve on the epoll portability abstraction, I would move to make that change to ofi_epoll itself, rather than creating a new abstraction over it.